peps/pep-0667.rst

```diff
@@ -371,6 +371,70 @@ C API
 As with all functions that return a borrowed reference, care must be taken to
 ensure that the reference is not used beyond the lifetime of the object.
 
+Impact on ``exec()``
+====================
+
+Even though this PEP does not modify ``exec()`` directly, the semantic change
+to ``locals()`` impacts the behavior of ``exec()``.
+
+``exec(object)`` has been equivalent to ``exec(object, globals(), locals())``
+for a long time, but it's not officially documented. We plan to keep this
+relation, which means existing code using ``exec()`` will see some different
+behaviors.
+
+The core difference is that the current implementation returns the same dict
+when ``locals()`` is called multiple times in function scope, so the following
+code magically works::
+
+    def f():
+        exec('a = 0')  # equivalent to exec('a = 0', globals(), locals())
+        exec('print(a)')  # equivalent to exec('print(a)', globals(), locals())
+        print(locals())  # {'a': 0}
+        # However, print(a) will not work here
+    f()
+
+``locals()`` in a function currently returns the same persisted dict for every call. This allows stashing
+extra "fake locals" in that dict, which aren't real locals known by the compiler, but can still be seen in
```

Comment:
```suggestion
extra "fake locals", which are values that aren't real locals known by the compiler. These "fake locals" can be seen in
```

Reply:
I reworded this paragraph in a way that I think covers this, so please readd the comment if it still applies.